// @nkwa-pay/sdk/dist/esm/models/errors/httperror.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { PayError } from "./payerror.js";
export class HttpError extends PayError {
    constructor(err, httpMeta) {
        const message = err.message || `API error occurred: ${JSON.stringify(err)}`;
        super(message, httpMeta);
        this.data$ = err;
        if (err.details != null)
            this.details = err.details;
        this.name = "HttpError";
    }
}
/** @internal */
export const HttpError$inboundSchema = z.object({
    details: z.array(z.any()).optional(),
    message: z.string().optional(),
    statusCode: z.number().int().optional(),
    request$: z.instanceof(Request),
    response$: z.instanceof(Response),
    body$: z.string(),
})
    .transform((v) => {
    return new HttpError(v, {
        request: v.request$,
        response: v.response$,
        body: v.body$,
    });
});
/** @internal */
export const HttpError$outboundSchema = z.instanceof(HttpError)
    .transform(v => v.data$)
    .pipe(z.object({
    details: z.array(z.any()).optional(),
    message: z.string().optional(),
    statusCode: z.number().int().optional(),
}));
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var HttpError$;
(function (HttpError$) {
    /** @deprecated use `HttpError$inboundSchema` instead. */
    HttpError$.inboundSchema = HttpError$inboundSchema;
    /** @deprecated use `HttpError$outboundSchema` instead. */
    HttpError$.outboundSchema = HttpError$outboundSchema;
})(HttpError$ || (HttpError$ = {}));
