// @nkwa-pay/sdk/dist/commonjs/models/components/availability.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || (function () {
    var ownKeys = function(o) {
        ownKeys = Object.getOwnPropertyNames || function (o) {
            var ar = [];
            for (var k in o) if (Object.prototype.hasOwnProperty.call(o, k)) ar[ar.length] = k;
            return ar;
        };
        return ownKeys(o);
    };
    return function (mod) {
        if (mod && mod.__esModule) return mod;
        var result = {};
        if (mod != null) for (var k = ownKeys(mod), i = 0; i < k.length; i++) if (k[i] !== "default") __createBinding(result, mod, k[i]);
        __setModuleDefault(result, mod);
        return result;
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.Availability$ = exports.Availability$outboundSchema = exports.Availability$inboundSchema = exports.Operation$ = exports.Operation$outboundSchema = exports.Operation$inboundSchema = exports.Status$ = exports.Status$outboundSchema = exports.Status$inboundSchema = exports.Type$ = exports.Type$outboundSchema = exports.Type$inboundSchema = exports.Operator$ = exports.Operator$outboundSchema = exports.Operator$inboundSchema = exports.Status = exports.Type = exports.Operator = void 0;
exports.operationToJSON = operationToJSON;
exports.operationFromJSON = operationFromJSON;
exports.availabilityToJSON = availabilityToJSON;
exports.availabilityFromJSON = availabilityFromJSON;
const z = __importStar(require("zod"));
const schemas_js_1 = require("../../lib/schemas.js");
exports.Operator = {
    Mtn: "mtn",
    Orange: "orange",
};
exports.Type = {
    Collection: "collection",
    Disbursement: "disbursement",
};
exports.Status = {
    Operational: "OPERATIONAL",
    Suspended: "SUSPENDED",
};
/** @internal */
exports.Operator$inboundSchema = z
    .nativeEnum(exports.Operator);
/** @internal */
exports.Operator$outboundSchema = exports.Operator$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Operator$;
(function (Operator$) {
    /** @deprecated use `Operator$inboundSchema` instead. */
    Operator$.inboundSchema = exports.Operator$inboundSchema;
    /** @deprecated use `Operator$outboundSchema` instead. */
    Operator$.outboundSchema = exports.Operator$outboundSchema;
})(Operator$ || (exports.Operator$ = Operator$ = {}));
/** @internal */
exports.Type$inboundSchema = z.nativeEnum(exports.Type);
/** @internal */
exports.Type$outboundSchema = exports.Type$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Type$;
(function (Type$) {
    /** @deprecated use `Type$inboundSchema` instead. */
    Type$.inboundSchema = exports.Type$inboundSchema;
    /** @deprecated use `Type$outboundSchema` instead. */
    Type$.outboundSchema = exports.Type$outboundSchema;
})(Type$ || (exports.Type$ = Type$ = {}));
/** @internal */
exports.Status$inboundSchema = z
    .nativeEnum(exports.Status);
/** @internal */
exports.Status$outboundSchema = exports.Status$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Status$;
(function (Status$) {
    /** @deprecated use `Status$inboundSchema` instead. */
    Status$.inboundSchema = exports.Status$inboundSchema;
    /** @deprecated use `Status$outboundSchema` instead. */
    Status$.outboundSchema = exports.Status$outboundSchema;
})(Status$ || (exports.Status$ = Status$ = {}));
/** @internal */
exports.Operation$inboundSchema = z.object({
    type: exports.Type$inboundSchema.optional(),
    status: exports.Status$inboundSchema.optional(),
});
/** @internal */
exports.Operation$outboundSchema = z.object({
    type: exports.Type$outboundSchema.optional(),
    status: exports.Status$outboundSchema.optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Operation$;
(function (Operation$) {
    /** @deprecated use `Operation$inboundSchema` instead. */
    Operation$.inboundSchema = exports.Operation$inboundSchema;
    /** @deprecated use `Operation$outboundSchema` instead. */
    Operation$.outboundSchema = exports.Operation$outboundSchema;
})(Operation$ || (exports.Operation$ = Operation$ = {}));
function operationToJSON(operation) {
    return JSON.stringify(exports.Operation$outboundSchema.parse(operation));
}
function operationFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Operation$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Operation' from JSON`);
}
/** @internal */
exports.Availability$inboundSchema = z.object({
    operator: exports.Operator$inboundSchema.optional(),
    operation: z.lazy(() => exports.Operation$inboundSchema).optional(),
});
/** @internal */
exports.Availability$outboundSchema = z.object({
    operator: exports.Operator$outboundSchema.optional(),
    operation: z.lazy(() => exports.Operation$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
var Availability$;
(function (Availability$) {
    /** @deprecated use `Availability$inboundSchema` instead. */
    Availability$.inboundSchema = exports.Availability$inboundSchema;
    /** @deprecated use `Availability$outboundSchema` instead. */
    Availability$.outboundSchema = exports.Availability$outboundSchema;
})(Availability$ || (exports.Availability$ = Availability$ = {}));
function availabilityToJSON(availability) {
    return JSON.stringify(exports.Availability$outboundSchema.parse(availability));
}
function availabilityFromJSON(jsonString) {
    return (0, schemas_js_1.safeParse)(jsonString, (x) => exports.Availability$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Availability' from JSON`);
}
