// @nkwa-pay/sdk/dist/commonjs/mcp-server/server.js
"use strict";
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.createMCPServer = createMCPServer;
const mcp_js_1 = require("@modelcontextprotocol/sdk/server/mcp.js");
const core_js_1 = require("../core.js");
const prompts_js_1 = require("./prompts.js");
const resources_js_1 = require("./resources.js");
const tools_js_1 = require("./tools.js");
const availabilityCheck_js_1 = require("./tools/availabilityCheck.js");
const paymentsCollect_js_1 = require("./tools/paymentsCollect.js");
const paymentsDisburse_js_1 = require("./tools/paymentsDisburse.js");
const paymentsGet_js_1 = require("./tools/paymentsGet.js");
function createMCPServer(deps) {
    const server = new mcp_js_1.McpServer({
        name: "Pay",
        version: "0.2.2",
    });
    const client = new core_js_1.PayCore({
        apiKeyAuth: deps.apiKeyAuth,
        serverURL: deps.serverURL,
        serverIdx: deps.serverIdx,
    });
    const scopes = new Set(deps.scopes);
    const allowedTools = deps.allowedTools && new Set(deps.allowedTools);
    const tool = (0, tools_js_1.createRegisterTool)(deps.logger, server, client, scopes, allowedTools);
    const resource = (0, resources_js_1.createRegisterResource)(deps.logger, server, client, scopes);
    const resourceTemplate = (0, resources_js_1.createRegisterResourceTemplate)(deps.logger, server, client, scopes);
    const prompt = (0, prompts_js_1.createRegisterPrompt)(deps.logger, server, client, scopes);
    const register = { tool, resource, resourceTemplate, prompt };
    void register; // suppress unused warnings
    tool(paymentsGet_js_1.tool$paymentsGet);
    tool(paymentsCollect_js_1.tool$paymentsCollect);
    tool(paymentsDisburse_js_1.tool$paymentsDisburse);
    tool(availabilityCheck_js_1.tool$availabilityCheck);
    return server;
}
