// @nkwa-pay/sdk/dist/esm/models/components/availability.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
export const Operator = {
    Mtn: "mtn",
    Orange: "orange",
};
export const Type = {
    Collection: "collection",
    Disbursement: "disbursement",
};
export const Status = {
    Operational: "OPERATIONAL",
    Suspended: "SUSPENDED",
};
/** @internal */
export const Operator$inboundSchema = z
    .nativeEnum(Operator);
/** @internal */
export const Operator$outboundSchema = Operator$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Operator$;
(function (Operator$) {
    /** @deprecated use `Operator$inboundSchema` instead. */
    Operator$.inboundSchema = Operator$inboundSchema;
    /** @deprecated use `Operator$outboundSchema` instead. */
    Operator$.outboundSchema = Operator$outboundSchema;
})(Operator$ || (Operator$ = {}));
/** @internal */
export const Type$inboundSchema = z.nativeEnum(Type);
/** @internal */
export const Type$outboundSchema = Type$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Type$;
(function (Type$) {
    /** @deprecated use `Type$inboundSchema` instead. */
    Type$.inboundSchema = Type$inboundSchema;
    /** @deprecated use `Type$outboundSchema` instead. */
    Type$.outboundSchema = Type$outboundSchema;
})(Type$ || (Type$ = {}));
/** @internal */
export const Status$inboundSchema = z
    .nativeEnum(Status);
/** @internal */
export const Status$outboundSchema = Status$inboundSchema;
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Status$;
(function (Status$) {
    /** @deprecated use `Status$inboundSchema` instead. */
    Status$.inboundSchema = Status$inboundSchema;
    /** @deprecated use `Status$outboundSchema` instead. */
    Status$.outboundSchema = Status$outboundSchema;
})(Status$ || (Status$ = {}));
/** @internal */
export const Operation$inboundSchema = z.object({
    type: Type$inboundSchema.optional(),
    status: Status$inboundSchema.optional(),
});
/** @internal */
export const Operation$outboundSchema = z.object({
    type: Type$outboundSchema.optional(),
    status: Status$outboundSchema.optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Operation$;
(function (Operation$) {
    /** @deprecated use `Operation$inboundSchema` instead. */
    Operation$.inboundSchema = Operation$inboundSchema;
    /** @deprecated use `Operation$outboundSchema` instead. */
    Operation$.outboundSchema = Operation$outboundSchema;
})(Operation$ || (Operation$ = {}));
export function operationToJSON(operation) {
    return JSON.stringify(Operation$outboundSchema.parse(operation));
}
export function operationFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Operation$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Operation' from JSON`);
}
/** @internal */
export const Availability$inboundSchema = z.object({
    operator: Operator$inboundSchema.optional(),
    operation: z.lazy(() => Operation$inboundSchema).optional(),
});
/** @internal */
export const Availability$outboundSchema = z.object({
    operator: Operator$outboundSchema.optional(),
    operation: z.lazy(() => Operation$outboundSchema).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Availability$;
(function (Availability$) {
    /** @deprecated use `Availability$inboundSchema` instead. */
    Availability$.inboundSchema = Availability$inboundSchema;
    /** @deprecated use `Availability$outboundSchema` instead. */
    Availability$.outboundSchema = Availability$outboundSchema;
})(Availability$ || (Availability$ = {}));
export function availabilityToJSON(availability) {
    return JSON.stringify(Availability$outboundSchema.parse(availability));
}
export function availabilityFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Availability$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Availability' from JSON`);
}
