// @nkwa-pay/sdk/dist/esm/models/components/paymentrequest.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const PaymentRequest$inboundSchema = z.object({
    amount: z.number().int(),
    phoneNumber: z.string(),
});
/** @internal */
export const PaymentRequest$outboundSchema = z.object({
    amount: z.number().int(),
    phoneNumber: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var PaymentRequest$;
(function (PaymentRequest$) {
    /** @deprecated use `PaymentRequest$inboundSchema` instead. */
    PaymentRequest$.inboundSchema = PaymentRequest$inboundSchema;
    /** @deprecated use `PaymentRequest$outboundSchema` instead. */
    PaymentRequest$.outboundSchema = PaymentRequest$outboundSchema;
})(PaymentRequest$ || (PaymentRequest$ = {}));
export function paymentRequestToJSON(paymentRequest) {
    return JSON.stringify(PaymentRequest$outboundSchema.parse(paymentRequest));
}
export function paymentRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => PaymentRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'PaymentRequest' from JSON`);
}
