// @nkwa-pay/sdk/dist/esm/models/components/payment.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
import { PaymentStatus$inboundSchema, PaymentStatus$outboundSchema, } from "./paymentstatus.js";
import { PaymentType$inboundSchema, PaymentType$outboundSchema, } from "./paymenttype.js";
import { TelecomOperator$inboundSchema, TelecomOperator$outboundSchema, } from "./telecomoperator.js";
/** @internal */
export const Payment$inboundSchema = z.object({
    amount: z.number().int().optional(),
    createdAt: z.string().datetime({ offset: true }).transform(v => new Date(v))
        .optional(),
    currency: z.string().optional(),
    description: z.string().optional(),
    fee: z.number().int().optional(),
    id: z.string().optional(),
    merchantId: z.number().int().optional(),
    paymentType: PaymentType$inboundSchema.optional(),
    phoneNumber: z.string().optional(),
    status: PaymentStatus$inboundSchema.optional(),
    telecomOperator: TelecomOperator$inboundSchema.optional(),
    updatedAt: z.string().datetime({ offset: true }).transform(v => new Date(v))
        .optional(),
});
/** @internal */
export const Payment$outboundSchema = z.object({
    amount: z.number().int().optional(),
    createdAt: z.date().transform(v => v.toISOString()).optional(),
    currency: z.string().optional(),
    description: z.string().optional(),
    fee: z.number().int().optional(),
    id: z.string().optional(),
    merchantId: z.number().int().optional(),
    paymentType: PaymentType$outboundSchema.optional(),
    phoneNumber: z.string().optional(),
    status: PaymentStatus$outboundSchema.optional(),
    telecomOperator: TelecomOperator$outboundSchema.optional(),
    updatedAt: z.date().transform(v => v.toISOString()).optional(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var Payment$;
(function (Payment$) {
    /** @deprecated use `Payment$inboundSchema` instead. */
    Payment$.inboundSchema = Payment$inboundSchema;
    /** @deprecated use `Payment$outboundSchema` instead. */
    Payment$.outboundSchema = Payment$outboundSchema;
})(Payment$ || (Payment$ = {}));
export function paymentToJSON(payment) {
    return JSON.stringify(Payment$outboundSchema.parse(payment));
}
export function paymentFromJSON(jsonString) {
    return safeParse(jsonString, (x) => Payment$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'Payment' from JSON`);
}
