// @nkwa-pay/sdk/dist/esm/models/operations/getpaymentsid.js
/*
 * Code generated by Speakeasy (https://speakeasy.com). DO NOT EDIT.
 */
import * as z from "zod";
import { safeParse } from "../../lib/schemas.js";
/** @internal */
export const GetPaymentsIdRequest$inboundSchema = z.object({
    id: z.string(),
});
/** @internal */
export const GetPaymentsIdRequest$outboundSchema = z.object({
    id: z.string(),
});
/**
 * @internal
 * @deprecated This namespace will be removed in future versions. Use schemas and types that are exported directly from this module.
 */
export var GetPaymentsIdRequest$;
(function (GetPaymentsIdRequest$) {
    /** @deprecated use `GetPaymentsIdRequest$inboundSchema` instead. */
    GetPaymentsIdRequest$.inboundSchema = GetPaymentsIdRequest$inboundSchema;
    /** @deprecated use `GetPaymentsIdRequest$outboundSchema` instead. */
    GetPaymentsIdRequest$.outboundSchema = GetPaymentsIdRequest$outboundSchema;
})(GetPaymentsIdRequest$ || (GetPaymentsIdRequest$ = {}));
export function getPaymentsIdRequestToJSON(getPaymentsIdRequest) {
    return JSON.stringify(GetPaymentsIdRequest$outboundSchema.parse(getPaymentsIdRequest));
}
export function getPaymentsIdRequestFromJSON(jsonString) {
    return safeParse(jsonString, (x) => GetPaymentsIdRequest$inboundSchema.parse(JSON.parse(x)), `Failed to parse 'GetPaymentsIdRequest' from JSON`);
}
